File: 2452-words-within-two-edits-of-dictionary/2452-words-within-two-edits-of-dictionary.java
class Solution {
    int hamming(String s1, String s2) {
        int h = 0;
        for (int i = 0; i < s1.length(); ++i) {
            h += s1.charAt(i) != s2.charAt(i) ? 1 : 0;
        }
        return h;
    };

    public List<String> twoEditWords(String[] queries, String[] dictionary) {
        ArrayList<String> words = new ArrayList<>();
        for (String q : queries)
            for (String d : dictionary)
                if (hamming(q, d) <= 2) {
                    words.add(q);
                    break;
                }
        return words;
    }
}

File: 2452-words-within-two-edits-of-dictionary/2452-words-within-two-edits-of-dictionary.cpp
class Solution {
public:
    vector<string> twoEditWords(vector<string>& queries,vector<string>& dictionary) {
        auto hamming = [](string& s1, string& s2) -> int {
            int h = 0;
            for (int i = 0; i < s1.size(); ++i)
                h += s1[i] != s2[i];
            return h;
        };

        vector<string> words;
        for (string& q : queries)
            for (string& d : dictionary)
                if (hamming(q, d) <= 2) {
                    words.push_back(q);
                    break;
                }

        return words;
    }
};